phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -260,7 +260,9 @@ void commitBatch(Region region, List<Mutation> mutations, long blockingMemstoreS
        Mutation[] mutationArray = new Mutation[mutations.size()];
       // When memstore size reaches blockingMemstoreSize we are waiting 3 seconds for the
       // flush happen which decrease the memstore size and then writes allowed on the region.
-      for (int i = 0; blockingMemstoreSize > 0 && region.getMemStoreHeapSize() > blockingMemstoreSize && i < 30; i++) {
+      for (int i = 0; blockingMemstoreSize > 0
+              && region.getMemStoreHeapSize() + region.getMemStoreOffHeapSize() > blockingMemstoreSize
```

Comment:
If we're including offheap memstore size, do we need to be calculating the `blockingMemstoreSize` differently?

It looks like we define it to be the table's `flushSize * (memstore_block_multiplier-1)`. I think this is still fine -- we're not changing how much memstore we hold on to, just whether that's on or off heap, right?

Reply:
I originally thought that this bug was introduced when we adopted to the API change in HBASE-18294, but on closer examination it it was fine. 
It was broken recently by the PHOENIX-6067 indexing rebase. 

We are really just restoring the HBase 2 semantics that were broken by PHOENIX-6067.